File: backend/sdk/java/src/main/java/com/waas/sdk/models/CreateEndpointRequest.java
package com.waas.sdk.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.Map;
import java.util.UUID;

/**
 * Request to create a webhook endpoint.
 */
public class CreateEndpointRequest {
    private String url;
    
    @JsonProperty("custom_headers")
    private Map<String, String> customHeaders;
    
    @JsonProperty("retry_config")
    private RetryConfiguration retryConfig;

    private CreateEndpointRequest(Builder builder) {
        this.url = builder.url;
        this.customHeaders = builder.customHeaders;
        this.retryConfig = builder.retryConfig;
    }

    public static Builder builder() {
        return new Builder();
    }

    public String getUrl() { return url; }
    public Map<String, String> getCustomHeaders() { return customHeaders; }
    public RetryConfiguration getRetryConfig() { return retryConfig; }

    public static class Builder {
        private String url;
        private Map<String, String> customHeaders;
        private RetryConfiguration retryConfig;

        public Builder url(String url) {
            this.url = url;
            return this;
        }

        public Builder customHeaders(Map<String, String> customHeaders) {
            this.customHeaders = customHeaders;
            return this;
        }

        public Builder retryConfig(RetryConfiguration retryConfig) {
            this.retryConfig = retryConfig;
            return this;
        }

        public CreateEndpointRequest build() {
            if (url == null || url.isEmpty()) {
                throw new IllegalArgumentException("URL is required");
            }
            return new CreateEndpointRequest(this);
        }
    }
}

File: backend/sdk/php/src/Models/CreateEndpointRequest.php
<?php

declare(strict_types=1);

namespace WAAS\Models;

class CreateEndpointRequest
{
    /**
     * @param array<string, int>|RetryConfiguration|null $retryConfig
     * @param array<string, string>|null $customHeaders
     */
    public function __construct(
        public readonly string $url,
        public readonly array|RetryConfiguration|null $retryConfig = null,
        public readonly ?array $customHeaders = null,
    ) {}

    /**
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        $data = ['url' => $this->url];

        if ($this->retryConfig !== null) {
            $data['retry_config'] = $this->retryConfig instanceof RetryConfiguration 
                ? $this->retryConfig->toArray() 
                : $this->retryConfig;
        }

        if ($this->customHeaders !== null) {
            $data['custom_headers'] = $this->customHeaders;
        }

        return $data;
    }
}
